articles/quickstart/native/android-vnext/01-user-metadata.md

```diff
@@ -0,0 +1,127 @@
+* add a call to `withAudience` and pass the audience for the Auth0 Management API: `"https://${account.namespace}/api/v2"`.
+
+```kotlin
+WebAuthProvider.login(account)
+  .withScheme("demo")
+  // modify the scopes to enable read and write of user_metadata
+  .withScope("openid profile email read:current_user update:current_user_metadata")
+  // specify the audience for the Auth0 Management API  
+  .withAudience("https://${account.namespace}/api/v2/")
+  .start(this, /* .. callback .. */)
+```
+
+:::note
+In the case of the Auth0 Management API, the `read:current_user` and `update:current_user_metadata` scopes let you get an access token that can retrieve user details and update the user's information. In the case of your APIs, you'll define custom [API scopes](https://auth0.com/docs/scopes/current/api-scopes) to implement access control, and you'll identify them in the calls that your client applications make to that API.
+
+In addition, in the case of the Auth0 Management API, the audience is `https://${account.namespace}/api/v2/`. In the case of your APIs, you create an _Identifier_ value that serves as the _Audience_ value whenever you [set up an API](https://auth0.com/docs/getting-started/set-up-api) with Auth0.
+:::
+
+Once you have a `UserProfile` instance (retrived by calling `userInfo` from the previous chapter), get the user's ID and then use the `UsersAPIClient` to get the full profile for the user in another `UserProfile` instance. This profile will include all of the user profile information, as well as additional fields such as `user_metadata` and `app_metadata`.
+
+:::note
+Fields such as `user_metadata` and `app_metadata` are not available on the user profile that is returned by calling `userInfo`, as these fields are proprietary fields to Auth0 and not part of the OIDC spec. You must call the [Auth0 Management API](https://auth0.com/docs/users/manage-user-metadata#management-api) to get these fields, which is what you are doing by using the `UsersAPIClient` in the snippet below.
+:::
+
+The metadata can then be retrieved by using the `getUserMetadata` method on the `UserProfile` object.
+
+```kotlin
+// Get the user ID and call the full getUser Management API endpoint, to retrieve the full profile information
+profile?.getId()?.let { userId ->
+    // Create the user API client using the account details and the access token from Credentials
+    val usersClient = UsersAPIClient(account, accessToken)
+
+    // Get the full user profile
+    usersClient
+        .getProfile(userId)
+        .start(object: Callback<UserProfile, ManagementException> {
+            override fun onFailure(exception: ManagementException) {
+                // Something went wrong!
+            }
+
+            override fun onSuccess(profile: UserProfile?) {
+                // Retrieve the "country" field, if one appears in the metadata
+                val country = profile?.getUserMetadata()?.get("country") as String? ?: ""
```

Comment:
maybe we could specify the val type instead of using the explicit cast. Also if the text is null that would be OK as we are saying the type is optional. Is for the caller to check.
```suggestion
                val country: String? = profile?.getUserMetadata()?.get("country")
```